pkg/sql/crdb_internal.go

```diff
@@ -867,6 +868,46 @@ func populateSessionsTable(
 	return nil
 }
 
+// crdbInternalLocalMetricsTable exposes a snapshot of the metrics on the
+// current node.
+var crdbInternalLocalMetricsTable = virtualSchemaTable{
+	schema: `CREATE TABLE crdb_internal.node_metrics (
+  store_id 	         INT NULL,       -- the store, if any, to which this metric belongs
+  name               STRING,         -- name of the metric
+  value							 FLOAT           -- value of the metric
+);`,
+
+	populate: func(ctx context.Context, p *planner, _ *DatabaseDescriptor, addRow func(...tree.Datum) error) error {
+		if err := p.RequireSuperUser(ctx, "read crdb_internal.node_metrics"); err != nil {
+			return err
+		}
+
+		mr := p.ExecCfg().MetricsRecorder
+		if mr == nil {
+			return nil
```

Comment:
Shouldn't we return an error if we're unable to return the requested information?